cpp/include/cuml/decomposition/kpca.hpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2018-2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "params.hpp"
+
+namespace raft {
+class handle_t;
+}
+
+namespace ML {
+
+void kpcaFit(raft::handle_t &handle, float *input, float *eigenvectors, float *eigenvalues, const paramsKPCA &prms);
+
+void kpcaFit(raft::handle_t &handle, double *input, double *eigenvectors,
+             double *eigenvalues, const paramsKPCA &prms);
+
+void kpcaTransform(raft::handle_t &handle, float *input, float *eigenvectors, float *eigenvalues,
+                   float *trans_input, const paramsKPCA &prms);
+
+void kpcaTransform(raft::handle_t &handle, double *input, double *eigenvectors, double *eigenvalues,
+                   double *trans_input, const paramsKPCA &prms);
+
+void kpcaTransform2(const raft::handle_t &handle, float *fit_input, float *input,
```

Comment:
Will refactor this API before moving PR out of draft. Currently rewriting `kpcaTransform()`, so the `test_kpca_fit_then_transform_on_test_train_split()` pytest will pass. Plan is to put the existing functionality in `kpcaTransform()` into a single `kpcaFitTransform()` function and promote `kpcaTransform2()` to `kpcaTransform()` once it's done.